### DISCUSSION TITLE
Energy bias correction

### DISCUSSION THREAD

**Discussion by dipierr** [gammapy/gammapy#4279] [Help]
Dear gammapy community,
   we have a question concerning the Energy bias handling.
We are using gammapy 0.20.1 providing DL3+IRFs input files. We have seen that there is the SafeMaskMaker, which allows to select a mask considering, for instance, a settable energy bias limit. Is there also a function to correct the reconstructed energy for the known energy bias? Is the energy automatically corrected or it is not corrected and we could just select the events accordingly to the quoted mask?

**Comment by registerrier** [comment]
Hello @dipierr . 
When computing the predicted number of counts in a `Dataset`, the energy dispersion is fully taken into account. So if the DL3 IRF energy dispersion contains a bias, it will be taken into account when evaluating the models.
No correction is applied to the reconstructed energy so that all values in `energy` (i.e. reconstructed energy) will remain biased though.  
The `mask_safe` that is built is defined in reconstructed energy. At the moment, the safe_mask determines the range in true energy in which the bias stays within some (user defined) limits and it assumes the valid range in reconstructed energy should be the same. This is obviously not really correct (e.g. because of the bias) and a better approach would be to assume a given spectrum and determine the valid range based on the probability of a given reco energy knowing the true energy.

**Reply by dipierr** [reply]
Dear @registerrier , many thanks for the clear answer.

> No correction is applied to the reconstructed energy so that all values in `energy` (i.e. reconstructed energy) will remain biased though.

Is there a reason for not correcting also the energy to bias-corrected energy?

**Reply by bkhelifi** [reply]
This is up to the IRFs experts to decide what is the best algorithms. They could add indeed a correction scheme that depends on the azimuth, the zenith angle, the offset, the true energy (and event types)... At the level of Gammapy, we are reading the IRFS and use them as they are.

**Reply by dipierr** [reply]
@registerrier , @bkhelifi  I don't fully get this point. Since gammapy uses correctly the energy dispersion to calculate the predicted number of counts in an energy bin, taking into account the E bias, those Energy bins should be in E "true" (bias corrected), so it should be also possible to keep the "true" energy scale for the x-axis. I still don't understand why then the "energy" is still the reconstructed energy while it would be more natural to show fluxes in bias-corrected Energy scale.

**Reply by bkhelifi** [reply]
Not exactly... Even if the bias is corrected, the resolution can not. And because of it, a given event at Ereco can come from different Etrue. This is no way to know what is the real Etrue, one can only derive a probability distribution in Etrue as a function of a fixed Ereco.

**Reply by dipierr** [reply]
This is clear, the resolution can not be corrected and should be taken into account differently (as you said), still I think that an average correction (Ebias) would be better than no-correction.

**Reply by bkhelifi** [reply]
And to go further, what is really the best is to not publish on the same figure quantities with different axis type;)
